src/modm/architecture/interface/spi_master.hpp

```diff
@@ -103,7 +103,7 @@ class SpiMaster : public ::modm::PeripheralDriver, public Spi
 	 * @return	received data
 	 */
 	static uint8_t
-	transferBlocking(uint8_t data);
+	transferBlocking(const uint8_t data);
```

Comment:
This is passed by value, so it's irrelevant if it's const. It also will cause issues, since some of the drivers modify the copy of the input data.

Reply:
Both function declarations are exactly equivalent, it only matters in the definition. The `const` is entirely ignored by the compiler in declaration. const by-value parameters are not generally recommended. Not so much here, but in other cases with non-trivial data types `const` inhibits move.  For example https://quuxplusone.github.io/blog/2019/01/03/const-is-a-contract/ has more discussions on that subject.

Reply:
Yeah, I remember falling into this trap when trying to have an overloaded function with and without const. Doesn't work, since it's ignored in the signature…

Reply:
You are right, I was focused on the pointers and didn't thought about the values.